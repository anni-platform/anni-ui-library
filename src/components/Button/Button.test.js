import React from 'react';
import renderer from 'react-test-renderer';
import defaultTheme from 'themes/default';
import { Button } from '.';

test('Button default', () => {
  expect(
    renderer
      .create(<Button theme={defaultTheme}>Perform action</Button>)
      .toJSON()
  ).toMatchSnapshot();
});

test('Button primary', () => {
  expect(
    renderer
      .create(
        <Button theme={defaultTheme} primary>
          Perform action
        </Button>
      )
      .toJSON()
  ).toMatchSnapshot();
});
