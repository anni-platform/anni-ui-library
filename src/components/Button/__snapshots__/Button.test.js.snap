// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  background: none;
  border: 2px solid #E2A480;
  border-radius: 80px;
  color: #1B1D1E;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Apercu Bold',sans-serif;
  font-size: 13px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1;
  outline: none;
  padding: 12px 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: 400ms cubic-bezier(0.19,1,0.22,1);
  transition: 400ms cubic-bezier(0.19,1,0.22,1);
  z-index: 2;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c0:active {
  -webkit-transform: scale(0.998);
  -ms-transform: scale(0.998);
  transform: scale(0.998);
}

.c0:focus,
.c0:hover {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
}

.c0:focus:disabled,
.c0:hover:disabled,
.c0:focus[disabled],
.c0:hover[disabled] {
  cursor: default;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<button
  className="c0"
>
  Perform action
</button>
`;

exports[`Button primary 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  background: #E2A480;
  border: 2px solid #E2A480;
  border-radius: 80px;
  color: #1B1D1E;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Apercu Bold',sans-serif;
  font-size: 13px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1;
  outline: none;
  padding: 12px 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: 400ms cubic-bezier(0.19,1,0.22,1);
  transition: 400ms cubic-bezier(0.19,1,0.22,1);
  z-index: 2;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c0:active {
  -webkit-transform: scale(0.998);
  -ms-transform: scale(0.998);
  transform: scale(0.998);
}

.c0:focus,
.c0:hover {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
}

.c0:focus:disabled,
.c0:hover:disabled,
.c0:focus[disabled],
.c0:hover[disabled] {
  cursor: default;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<button
  className="c0"
>
  Perform action
</button>
`;
